validator/checkout.js

```diff
@@ -0,0 +1,107 @@
+	data.county = ( ! isEmpty( data.county ) ) ? data.county : '';
+	data.postCode = ( ! isEmpty( data.postCode ) ) ? data.postCode : '';
+	data.phone = ( ! isEmpty( data.phone ) ) ? data.phone : '';
+	data.email = ( ! isEmpty( data.email ) ) ? data.email : '';
+	data.createAccount = ( ! isEmpty( data.createAccount ) ) ? data.createAccount : '';
+	data.orderNotes = ( ! isEmpty( data.orderNotes ) ) ? data.orderNotes : '';
+	data.paymentMode = ( ! isEmpty( data.paymentMode ) ) ? data.paymentMode : '';
+
+	/**
+	 * Checks for error if required is true
+	 * and adds Error and Sanitized data to the errors and sanitizedData object
+	 *
+	 * @param {String} fieldName Field name e.g. First name, last name
+	 * @param {String} errorContent Error Content to be used in showing error e.g. First Name, Last Name
+	 * @param {Integer} min Minimum characters required
+	 * @param {Integer} max Maximum characters required
+	 * @param {String } type Type e.g. email, phone etc.
+	 * @param {boolean} required Required if required is passed as false, it will not validate error and just do sanitization.
+	 */
+	const addErrorAndSanitizedData = ( fieldName, errorContent, min, max, type = '', required ) => {
+
+		const postCodeLocale = config.postCodeLocale ? config.postCodeLocale : '';
+		/**
+		 * Please note that this isEmpty() belongs to validator and not our custom function defined above.
+		 *
+		 * Check for error and if there is no error then sanitize data.
+		 */
+		if ( ! validator.isLength( data[ fieldName ], { min, max } ) ){
+			errors[ fieldName ] = `${errorContent} must be ${min} to ${max} characters`;
+		}
+
+		if ( 'email' === type && ! validator.isEmail( data[ fieldName ] ) ){
+			errors[ fieldName ] = `${errorContent} is not valid`;
+		}
+
+		if ( 'phone' === type && ! validator.isMobilePhone( data[ fieldName ] ) ) {
+			errors[ fieldName ] = `${errorContent} is not valid`;
+		}
+
+		if ( 'postCode' === type && postCodeLocale && ! validator.isPostalCode( data[ fieldName ], postCodeLocale ) ) {
+			errors[ fieldName ] = `${errorContent} is not valid`;
+		}
+
+		if ( required && validator.isEmpty( data[ fieldName ] ) ) {
+			errors[ fieldName ] = `${errorContent} is required`;
+		}
+
+
+		// If no errors
+		if ( ! errors[ fieldName ] ) {
+			sanitizedData[ fieldName ] = validator.trim( data[ fieldName ] );
+			sanitizedData[ fieldName ] = ( 'email' === type ) ? validator.normalizeEmail( sanitizedData[ fieldName ] ) : sanitizedData[ fieldName ];
+			sanitizedData[ fieldName ] = validator.escape( sanitizedData[ fieldName ] );
+		}
+
+	};
+
+	addErrorAndSanitizedData( 'firstName', 'First name', 2, 35, 'string', true );
+	addErrorAndSanitizedData( 'lastName', 'Last name', 2, 35, 'string', true );
+	addErrorAndSanitizedData( 'companyName', 'Company Name', 0, 35, 'string', false );
+	addErrorAndSanitizedData( 'country', 'Country name', 2, 55, 'string', true );
+	addErrorAndSanitizedData( 'streetAddressOne', 'Street address line 1', 35, 100,'string',true );
```

Comment:
Can we change this to `addErrorAndSanitizedData( 'streetAddressOne', 'Street address line 1', 5, 100,'string',true );`? Because 35 is too long for most address line 1